### PR TITLE
Add comprehensive tests for TagsService (#7)

### LINKED ISSUE #7: Add tests for `TagsService`
[`TagsService`](https://github.com/deyna256/betterforces/blob/main/backend/domain/services/tags_service.py) has no test coverage.

**Methods to test:**
- `analyze_tags()` - calculates average/median ratings per tag
- `get_weak_tags()` - filters tags below threshold

**Edge cases for example:**
- Empty submissions
- Problems without tags
- Threshold filtering logic

### PR DESCRIPTION
**Pull request by EdoAbarca**
- Add test coverage for analyze_tags() method
- Add test coverage for get_weak_tags() method
- Test edge cases: empty submissions, no ratings, no tags
- Test deduplication logic
- Test statistical calculations (average, median)
- Test threshold filtering for weak tags
- Test sorting behavior for both methods

All tests pass with 100% coverage of tags_service.py and 98% backend coverage.

Resolve #7

### CODE REVIEW

File: backend/tests/unit/tags_service/test_analysis.py [hunk 1]
```diff
@@ -0,0 +1,649 @@
...
+
+from backend.domain.models.codeforces import SubmissionStatus
+from backend.domain.services.tags_service import TagsService
+
+
+def test_empty_submissions() -> None:
+    """
+    Verifies that analyzing an empty list of submissions results in a valid,
+    empty analysis object with zero counts.
+    """
```
Comment (deyna256) [inline backend/tests/unit/tags_service/test_analysis.py]: You haven't to add docstring for a test, if you move it to the right place and name clear

**Review by deyna256 — CHANGES_REQUESTED** [review]
@EdoAbarca good job! but need to fix some issues: redundant comments and docstrings

File: backend/tests/unit/tags_service/test_analysis.py (hunk 1 above)
Comment (EdoAbarca) [reply backend/tests/unit/tags_service/test_analysis.py]: Thanks, after reading the document, I decided to also change the test suite's titles, now explicitly describing what it's tested and in what conditions. Committed and ready for review 👍
